packages/server/modules/gatekeeper/graph/resolvers/index.ts

```diff
@@ -301,21 +306,46 @@ export = FF_GATEKEEPER_MODULE_ENABLED
           )
           const stripe = getStripeClient()
 
-          const countWorkspaceRole = countWorkspaceRoleWithOptionalProjectRoleFactory({
-            db
+          const currentPlan = await getWorkspacePlan({ workspaceId })
+          const upgradeWorkspaceSubscription =
+            currentPlan && isNewPlanType(currentPlan.name)
+              ? upgradeWorkspaceSubscriptionFactoryNew({
+                  getWorkspacePlan: getWorkspacePlanFactory({ db }),
+                  reconcileSubscriptionData: reconcileWorkspaceSubscriptionFactory({
+                    stripe
+                  }),
+                  countSeatsByTypeInWorkspace: countSeatsByTypeInWorkspaceFactory({
+                    db
+                  }),
+                  getWorkspaceSubscription: getWorkspaceSubscriptionFactory({ db }),
+                  getWorkspacePlanPriceId,
+                  getWorkspacePlanProductId,
+                  upsertWorkspacePlan: upsertPaidWorkspacePlanFactory({ db }),
+                  updateWorkspaceSubscription: upsertWorkspaceSubscriptionFactory({
+                    db
+                  })
+                })
+              : upgradeWorkspaceSubscriptionFactoryOld({
+                  getWorkspacePlan: getWorkspacePlanFactory({ db }),
+                  reconcileSubscriptionData: reconcileWorkspaceSubscriptionFactory({
+                    stripe
+                  }),
+                  countWorkspaceRole: countWorkspaceRoleWithOptionalProjectRoleFactory({
+                    db
+                  }),
+                  getWorkspaceSubscription: getWorkspaceSubscriptionFactory({ db }),
+                  getWorkspacePlanPriceId,
+                  getWorkspacePlanProductId,
+                  upsertWorkspacePlan: upsertPaidWorkspacePlanFactory({ db }),
+                  updateWorkspaceSubscription: upsertWorkspaceSubscriptionFactory({
+                    db
+                  })
+                })
+          await upgradeWorkspaceSubscription({
+            workspaceId,
+            targetPlan: workspacePlan as PaidWorkspacePlansNew, // This should not be casted and the cast will be removed once we will not support old plans anymore
```

Comment:
why is this cast necessary? both the old & new upgrade implementation expects targetPlan to be `PaidWorkspacePlans` not `PaidWorkspacePlansNew` which should work without the cast, no?

Reply:
actually no
the old should expect the old only, and the new should expect the new only.
I found the old implementation changed so that it also accepts the new plans, so I left it as it was.